game_data_cleaner/game_data_cleaner.py

```diff
@@ -0,0 +1,138 @@
+import pandas as pd
+import numpy as np
+import requests
+import regex as re
+import time
+import os
+import gc
+import json
+from datetime import datetime
+
+# from statistics import mean
+# NLP tools
+# import spacy
+
+# nlp = spacy.load("en_core_web_sm")
+# import re
+# import nltk
+# from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer
+# from nltk.tokenize import word_tokenize
+
+from utils.s3_file_handler import S3FileHandler
+from utils.local_file_handler import LocalFileHandler
+from processing_functions import integer_reduce
+from config import CONFIGS
+
+GAME_CONFIGS = CONFIGS["game"]
+
+
+class GameDataCleaner:
+
+    def __init__(self) -> None:
+        self.s3_file_handler = S3FileHandler()
+        self.local_handler = LocalFileHandler()
+
+    def cleaning_chain(self) -> pd.DataFrame:
+        games_df = self.load_games_data("data/game_dfs_dirty/games.pkl")
+        games_df = self._drop_duplicates(games_df)
+        games_df = self._drop_unneeded_columns(games_df)
+        games_df = self._clean_best_players(games_df)
+        games_df = self._add_binary_category_flags(games_df)
+        games_df = self._reduce_integers_for_memory_usage(games_df)
+        games_df = self._drop_unreleased(games_df)
+        games_df = self._set_missing_min_players(games_df)
+        games_df, themes_df = self._breakout_themes_df(games_df)
```

Comment:
I notice many of these steps start with a df.copy().  I think that uses more memory, and I don't think there's a reason to do that since we reassign `games_df` at every function call.